sql/catalyst/src/main/scala/org/apache/spark/sql/types/date/Date.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types.date
+
+import java.util.{Calendar, TimeZone}
+
+/**
+ * A mutable implementation of java.sql.Date that holds an Int for days since epoch, like Hive.
+ */
+final class Date extends Ordered[Date] with Serializable {
+  private var daysSinceEpoch: Int = 0
+
+  /**
+   * Set this Decimal to the given Long. Will have precision 20 and scale 0.
+   */
+  def set(days: Int): Date = {
+    this.daysSinceEpoch = days
+    this
+  }
+
+  /**
+   * Set this Decimal to the given Int. Will have precision 10 and scale 0.
+   */
+  def set(days: Long): Date = {
+    this.daysSinceEpoch = days.toInt
+    this
+  }
+
+  /**
+   * Set this Decimal to the given BigDecimal value, inheriting its precision and scale.
+   */
+  def set(date: Date): Date = {
+    this.daysSinceEpoch = date.daysSinceEpoch
+    this
+  }
+
+  def toDays: Int = {
+    this.daysSinceEpoch
+  }
+
+  def toJavaDate: java.sql.Date = {
+    new java.sql.Date(toLong)
+  }
+
+  override def toString: String = toJavaDate.toString()
+
+  def toLong: Long = {
```

Comment:
Can we remove toLong, toInt, toShort, toByte? I'm not sure whether they are useful. If they are, we can always add them in the future. Once they are added, we cannot remove them easily.